src/clj/rems/db/applications.clj

```diff
@@ -282,11 +283,15 @@
   :start (scheduler/start! reload-cache! (Duration/standardHours 1))
   :stop (scheduler/stop! all-applications-cache-reloader))
 
+(defn remove-application-data!
+  [app-id]
+  (db/delete-application-attachments! {:application app-id})
+  (db/delete-application-events! {:application app-id})
+  (db/delete-application! {:application app-id}))
+
 (defn delete-application! [app-id]
   (let [application (get-application app-id)]
     (assert (= :application.state/draft (:application/state application))
             (str "Tried to delete application " app-id " which is not a draft!")))
-  (db/delete-application-attachments! {:application app-id})
-  (db/delete-application-events! {:application app-id})
-  (db/delete-application! {:application app-id})
-  (reload-cache!))
+  (remove-application-data! app-id)
+  (reload-cache!))
```

Comment:
I'm thinking that this part is slightly wrong in that it reloads the cache (too slow). We should add a task about improving this. Also is it really a `db` layer concern or `service`?

Reply:
i would think `service` is concerned with core business logic, while `db` is concerned with underlying implementation, such as generating correct queries or caching results?

Reply:
I'm just not sure about the caching part. If it's completely transparent and performant, it matters less how it works. But here it was too slow.

Also if it just caches the `db` representation then it belongs there. But if it would cache the "joined" full representation it should be in the `service`?

Finally, there is an mismatch here between removing one application and updating all of the cache. That feels simplistic. 

We don't have to touch this in this PR, but it feels like something we need to clarify and refactor in the future. I can add a ticket about this.

Reply:
I added to the existing ticket https://github.com/CSCfi/rems/issues/2647